core/thread.c

```diff
@@ -55,25 +55,28 @@ const char *thread_getname(kernel_pid_t pid)
 #endif
 }
 
-void thread_zombify(void)
+int thread_zombify(void)
```

Comment:
Does `NORETURN` instead of `int` work?

Reply:
It has to return when `irq_is_in()` is true.